### PR TITLE
fix(components/action-bars): persist focus on summary action bar chevrons after animations

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

File: libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar-adapter.service.ts [hunk 1]
```diff
@@ -22,6 +22,14 @@ export class SkySummaryActionBarAdapterService {
     this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
   }
 
+  public focusChevron(chevronElement: ElementRef) {
+    /* sanity-check */
+    /* istanbul ignore else */
+    if (chevronElement) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar-adapter.service.ts]: In what scenario would the `chevronElement` be undefined?

Comment (Blackbaud-TrevorBurch) [reply libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar-adapter.service.ts]: I really shouldn’t. However, I did hit this on accident because I had forgotten to add a check for if the animation had fired when the element was destroyed (Angular just does that). 

I added a check to the animation handler so it doesn’t call this code in that case now, but I left the null check here just as a safety net since this is a public function that could be called with `undefined`.

Comment (Blackbaud-TrevorBurch) [reply libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar-adapter.service.ts]: I ignored the branch because there is no current path that leads to `undefined` being passed in and we don’t have any tests that call adapter methods directly.

File: libs/components/action-bars/package.json [hunk 2]
```diff
@@ -16,6 +16,7 @@
   },
   "homepage": "https://github.com/blackbaud/skyux#readme",
   "peerDependencies": {
+    "@angular/animations": "^12.2.16",
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/action-bars/package.json:19]: A note on this - this is not a breaking change even though it looks like it may be. I have to add this because I now import from the library. However, there is already a peer on `@skyux/animations` that had a peer of `@angular/animations` - so it isn't a breaking change as `@angular/animations` was already required by a required library.

File: libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar-adapter.service.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar-adapter.service.ts]: I'd either:
1. Remove the `undefined` check, or
2. Make the method's parameter `ElementRef | undefined`.

I'd personally prefer the first option, and add a check for undefined later when a specific scenario is encountered that can be unit tested.

File: libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.ts [hunk 3]
```diff
@@ -148,16 +153,26 @@ export class SkySummaryActionBarComponent implements AfterViewInit, OnDestroy {
...
+      if (
+        this.type === SkySummaryActionBarType.Page ||
+        this.type === SkySummaryActionBarType.Tab
+      ) {
+        this.adapterService.styleBodyElementForActionBar(this.elementRef);
+      }
+
+      // Ensure that the correct chevron is fully rendered prior to setting focus.
+      setTimeout(() => {
+        this.adapterService.focusChevron(this.chevronElementRef);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.ts:175]: Possible to write a unit test that covers this scenario? We have a lot of code being added that's not getting tested.

Comment (Blackbaud-TrevorBurch) [reply libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.ts:175]: Facepalm..... that was tired Trevor. Sorry about that. I'll get one added

File: libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar-adapter.service.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar-adapter.service.ts]: Removed

File: libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.spec.ts [hunk 4]
```diff
@@ -354,23 +376,56 @@ describe('Summary Action Bar component', () => {
...
-          .nativeElement.click();
+        clickExpandButton(debugElement);
         fixture.detectChanges();
         tick();
         fixture.detectChanges();
         expect(cmp.summaryActionBar.isSummaryCollapsed).toBeFalsy();
         expect(cmp.summaryActionBar.slideDirection).toBe('down');
       }));
+
+      it(`should move focus to the collapsed summary's chevron after collapsing`, async () => {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.spec.ts:393]: @Blackbaud-SteveBrush This and the next test are the new ones. Everything else here was general cleanup that I did as this file was hard to read in a lot of places and had various `any` and deprecation issues. I'm happy to move the refactors to a different PR if you'd like.

File: libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.ts:175]: Done

File: libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.spec.ts (hunk 4 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.spec.ts:393]: Love it.

**Review by Blackbaud-SteveBrush — APPROVED** [review]
